### DISCUSSION TITLE
Will increasing JDBC Connection timeout help?

### DISCUSSION THREAD

**Discussion by vladgardusseedblink** [keycloak/keycloak#26197] [Q&A]
Hi, I am using AWS to host Keycloak in a docker container with ECS and connected to a RDS database. For one environment I used RDS Aurora Serverless V1 cluster so that the database can scale to 0 when it is not used so that we don't pay the costs of the database being idle.

I also use the keycloak-js npm package in my frontend application. The issue is that sometimes when the database is scaled to 0 and a user tries to access our app, considering that it can take lets say 5-10 seconds for the cluster to spin up, they get redirected to keycloak but the network request to `/realms/realm-name/protocol/openid-connect/3p-cookies/step1.html` fails with error 500 and looking at the keycloak logs I see the following errors: 
- HTTP Request to /realms/realm-name/protocol/openid-connect/3p-cookies/step1.html failed, error id: 95da2922-5808-4db3-b9d4-43e15ba0c5cc-1: org.jboss.resteasy.spi.UnhandledException: javax.ws.rs.InternalServerErrorException: HTTP 500 Internal Server Error
- ERROR [org.keycloak.headers.DefaultSecurityHeadersProvider] (executor-thread-27) MediaType not set on path /realms/realm-name/protocol/openid-connect/3p-cookies/step1.html, with response status 500
- ERROR [org.keycloak.services.error.KeycloakErrorHandler] (executor-thread-27) Failed to create error page: javax.persistence.PersistenceException: org.hibernate.exception.GenericJDBCException: Unable to acquire JDBC Connection

Will increasing the JDBC connection timeout help me get over those errors? And if so, how can I increase this timeout? I searched in the keycloak docs but I was unable to find a variable name that would serve my use case.

**Comment by sschu** [comment]
You can probably set this https://quarkus.io/guides/all-config#quarkus-agroal_quarkus.datasource.jdbc.acquisition-timeout.
However, Keycloak typically also has some background DB jobs running so I am not sure it will run very well when it doesnt have the database available.
